java/org/apache/jasper/compiler/NewlineReductionServletWriter.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.jasper.compiler;
+
+import java.io.PrintWriter;
+
+/**
+ * This class filters duplicate newlines instructions from the compiler output,
+ * and therefore from the runtime JSP. The duplicates typically happen because
+ * the compiler has multiple branches that write them, but they operate
+ * independently and don't realize that the previous output was identical.
+ *
+ * Removing these lines makes the JSP more efficient by executing fewer operations during runtime.
+ *
+ * @author Engebretson, John
+ * @author Kamnani, Jatin
+ *
+ */
+public class NewlineReductionServletWriter extends ServletWriter {
+    private static final String NEWLINE_WRITE_TEXT = "out.write('\\n');";
```

Comment:
There are no calls that match this pattern in the Tomcat code base. Due to this point alone, I don't see the justification for this feature.
Skipping blank lines like this is going to break SMAP support. That isn't a show stopper but needs to be noted in the documentation.

Reply:
SMAP Support defaults to false, and this feature has a property which can be default to false as well. 
Can you provide me the links where I can write the information on this patch ? @markt-asf